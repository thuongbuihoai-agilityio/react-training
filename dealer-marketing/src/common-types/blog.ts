export interface Blogs {
  blogId: string;
  title: string;
  slug: string;
  createDate: any;
  expertId: string;
  image: {
    url: any;
    alt: string;
  };
  description: string;
}
