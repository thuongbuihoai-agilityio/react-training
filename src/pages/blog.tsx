import { loadPosts } from "./lib/load-posts"

// This function runs only on the server side
export async function getStaticProps() {
  const posts = await loadPosts()
  return { props: { posts } }
}