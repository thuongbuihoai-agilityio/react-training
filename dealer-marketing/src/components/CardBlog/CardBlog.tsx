import React from "react";
import { Blogs } from "@common-types/blog";
import ResearchSection from "@sections/ResearchSection/ResearchSection";
import styleCardBlog from "./cardBlog.module.css";

interface CartBlogProps {
  blog: Blogs;
}

const CartBlog: React.FC<CartBlogProps> = ({ blog }) => {
  return (
    <div className={styleCardBlog["cart-blog"]}>
      <ResearchSection
        blog={blog}
        content="left"
        layout="gird"
        isButton={false}
        imageSmall={true}
      />
    </div>
  );
};

export default CartBlog;
