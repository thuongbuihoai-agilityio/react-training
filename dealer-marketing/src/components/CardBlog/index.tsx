import React, { memo } from "react";
import { Blog, BlogLayoutType, BlogContentType } from "@common-types/blog";
import ResearchSection from "@sections/ResearchSection";
import { BLOG_MOCKING } from "@constants/blog";

interface CartBlogProps {
  blog: Blog;
}

const CardBlog: React.FC<CartBlogProps> = ({ blog = BLOG_MOCKING }) => (
  <ResearchSection
    blog={blog}
    content={BlogContentType.left}
    layout={BlogLayoutType.grid}
    isButton={false}
    imageSmall={true}
  />
);

export default memo(CardBlog);
