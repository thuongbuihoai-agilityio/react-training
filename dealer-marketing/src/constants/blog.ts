import { Blog } from "@common-types/blog";

const BLOG_MOCKING: Blog = {
  blogId: "1",
  title: "Exploring Influential and Impactful Automotive Advertising Campaigns",
  slug: "exploring-influential-and-impactful-automotive-advertising-campaigns",
  createDate: "July 28, 2022",
  expertId: "Donna Welker",
  image: {
    url: "/images/past-present-future.png",
    alt: "This is image past present future",
  },
  description:
    "Advertising has come a long way since its humble beginnings. There's no doubt about it â€“ marketing remains an ever-changing landscape: What worked a few years ago may not be effective today, and what's popular now may be out of style in a few months. Nowhere is this more relevant than in the automotive industry, where new campaigns hold the potential to make or break a company. This article will explore some disruptive and impactful automotive marketing campaigns throughout the past century. We'll explore what made them successful and see how they changed the marketing landscape forever.",
};

export { BLOG_MOCKING };
