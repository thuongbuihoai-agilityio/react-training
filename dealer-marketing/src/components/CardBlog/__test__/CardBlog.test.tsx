import { BlogContextProps } from "@common-types/blog";
import { BLOG_MOCKING, BLOG_MOCKING_LIST } from "@constants/blog";
import { BlogContext } from "@context/BlogContext";
import { render, screen } from "@testing-library/react";
import CardBlog from "../index";

const contextBlogMocking: BlogContextProps = {
  searchValue: "",
  setSearchValue: jest.fn(),
  blogs: BLOG_MOCKING_LIST,
  blogList: BLOG_MOCKING_LIST,
  errorCode: 0,
  setBlogs: jest.fn(),
  setErrorCode: jest.fn(),
  handleUpdateBlogs: jest.fn(),
};

describe("CardBog component", () => {
  test("Should render CardBog component", () => {
    render(<CardBlog blog={BLOG_MOCKING} />);
    const cardBog = screen.getByText(/RESEARCH & ANALYSIS/i);
    expect(cardBog).toBeInTheDocument();
  });

  test("Matches snapshot", () => {
    const { container } = render(
      <BlogContext.Provider value={contextBlogMocking}>
        <CardBlog blog={BLOG_MOCKING} />
      </BlogContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
