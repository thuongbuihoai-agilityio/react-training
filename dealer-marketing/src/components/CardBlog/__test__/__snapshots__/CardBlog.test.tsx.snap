// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBog component Matches snapshot 1`] = `
<div>
  <div
    data-testid="research-section"
  >
    <div>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27350%27%20height=%27210%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="This is image past present future"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/images/backgrounds/blur.jpg);"
        />
        <noscript />
      </span>
      <div>
        <p>
          Research & analysis
        </p>
        <div>
          <div>
            <p
              data-testid="text"
            >
              Exploring Influential and Impactful Automotive Advertising Campaigns
            </p>
          </div>
          <div>
            <p>
              By
               
              <span>
                Donna Welker
                 
              </span>
              - 
              <span>
                July 28, 2022
              </span>
            </p>
            <p
              data-testid="text"
            >
              Advertising has come a long way since its humble beginnings. There's no doubt about it â€“ marketing remains an ever-changing landscape: What worked a few years ago may not be effective today, and what's popular now may be out of style in a few months. Nowhere is this more relevant than in the automotive industry, where new campaigns hold the potential to make or break a company. This article will explore some disruptive and impactful automotive marketing campaigns throughout the past century. We'll explore what made them successful and see how they changed the marketing landscape forever.
            </p>
          </div>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
